src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -377,9 +377,13 @@ PeerImp::hasLedger (uint256 const& hash, std::uint32_t seq) const
             return true;
     }
 
-    if (seq >= app_.getNodeStore().earliestSeq())
-        return hasShard(
-            (seq - 1) / NodeStore::DatabaseShard::ledgersPerShardDefault);
+    if (auto shardStore = app_.getShardStore())
+    {
+        if (seq >= shardStore->earliestSeq())
+            return hasShard(shardStore->seqToShardIndex(seq));
+    }
+    else if (seq >= app_.getNodeStore().earliestSeq())
```

Comment:
Do we want the `else` here? I _think_ that we don't. As is, the code logic makes it so that if we have a shard store and the ledger isn't there we never even bother looking in the node store.

Reply:
Determing ledgers per shard is solely for a unit test purposes and not needed here. Simplfied to ```    return seq >= app_.getNodeStore().earliestSeq() &&
        hasShard(NodeStore::seqToShardIndex(seq));```